File: problems/002-two-sum/dart/clean.dart
dynamic solve(Map<String, dynamic> c) {
  final nums = (c['nums'] as List).cast<num>();
  final target = c['target'] as num;
  final seen = <num, int>{};
  for (var i = 0; i < nums.length; i++) {
    final need = target - nums[i];
    if (seen.containsKey(need)) return [seen[need], i];
    seen[nums[i]] = i;
  }
  return [];
}

File: problems/002-two-sum/typescript/clean.ts
export function solve(c: { nums: number[]; target: number }): number[] {
  const seen = new Map<number, number>();
  for (let i = 0; i < c.nums.length; i++) {
    const need = c.target - c.nums[i];
    if (seen.has(need)) return [seen.get(need)!, i];
    seen.set(c.nums[i], i);
  }
  return [];
}
